### PR TITLE
Add testbed for the awsxray exporter

### PR DESCRIPTION
**Pull request by darwhs**
**Description:** Created the testbed for the AWSXray Exporter 

**Link to tracking Issue:** open-telemetry/opentelemetry-collector-contrib#372
https://github.com/open-telemetry/opentelemetry-collector/pull/1288

**Testing:** This change must be merge only after merging the PR https://github.com/open-telemetry/opentelemetry-collector/pull/1288 as I have added new method in the mock_backend class of collector package

on my local I have tested this change by replacing the collector with my local version by adding below line in my mod file
`replace go.opentelemetry.io/collector v0.5.0 => ../../opentelemetry-collector`

below are result of my testing
```
❯ ./runtests.sh
=== RUN   TestMetric10kDPS
=== RUN   TestMetric10kDPS/OpenCensus
2020/07/08 19:03:49 Starting mock backend...
2020/07/08 19:03:49 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:03:49 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestMetric10kDPS/OpenCensus/agent.log
2020/07/08 19:03:49 Agent running, pid=95389
2020/07/08 19:03:49 Starting load generator at 10000 items/sec.
2020/07/08 19:03:52 Agent RAM (RES):   2 MiB, CPU: 0.0% | Sent:   185,500 items | Received:   185,500 items (61,852/sec)
2020/07/08 19:03:55 Agent RAM (RES):  33 MiB, CPU:26.2% | Sent:   395,500 items | Received:   394,800 items (65,813/sec)
2020/07/08 19:03:58 Agent RAM (RES):  34 MiB, CPU:24.6% | Sent:   605,500 items | Received:   605,500 items (67,279/sec)
2020/07/08 19:04:01 Agent RAM (RES):  34 MiB, CPU:24.0% | Sent:   815,500 items | Received:   814,800 items (67,905/sec)
2020/07/08 19:04:04 Agent RAM (RES):  34 MiB, CPU:24.7% | Sent: 1,025,500 items | Received: 1,025,500 items (68,371/sec)
2020/07/08 19:04:04 Stopped generator. Sent: 1,049,300 items
2020/07/08 19:04:04 Gracefully terminating Agent pid=95389, sending SIGTEM...
2020/07/08 19:04:04 Stopping process monitor.
2020/07/08 19:04:04 Agent process stopped, exit code=0
2020/07/08 19:04:04 Sent and received data matches.
2020/07/08 19:04:04 Stopping mock backend...
2020/07/08 19:04:04 Stopped backend. Received: 1,049,300 items (67,771/sec)
=== RUN   TestMetric10kDPS/Carbon
2020/07/08 19:04:04 Starting mock backend...
2020/07/08 19:04:04 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:04:04 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestMetric10kDPS/Carbon/agent.log
2020/07/08 19:04:04 Agent running, pid=95481
2020/07/08 19:04:04 Starting load generator at 10000 items/sec.
2020/07/08 19:04:07 Agent RAM (RES):  12 MiB, CPU: 0.0% | Sent:    87,500 items | Received:    76,709 items (25,580/sec)
2020/07/08 19:04:10 Agent RAM (RES):  35 MiB, CPU:162.2% | Sent:   167,300 items | Received:   156,241 items (26,045/sec)
2020/07/08 19:04:13 Agent RAM (RES):  35 MiB, CPU:166.4% | Sent:   250,600 items | Received:   239,457 items (26,610/sec)
2020/07/08 19:04:16 Agent RAM (RES):  35 MiB, CPU:170.8% | Sent:   331,100 items | Received:   320,993 items (26,752/sec)
2020/07/08 19:04:19 Agent RAM (RES):  35 MiB, CPU:170.2% | Sent:   415,100 items | Received:   403,599 items (26,908/sec)
2020/07/08 19:04:19 Stopped generator. Sent:   415,100 items
2020/07/08 19:04:20 Gracefully terminating Agent pid=95481, sending SIGTEM...
2020/07/08 19:04:20 Stopping process monitor.
2020/07/08 19:04:20 Agent process stopped, exit code=0
2020/07/08 19:04:20 Sent and received data matches.
2020/07/08 19:04:20 Stopping mock backend...
2020/07/08 19:04:20 Fatal error reported: accept tcp 127.0.0.1:62073: use of closed network connection
2020/07/08 19:04:20 Stopped backend. Received:   415,100 items (26,406/sec)
=== RUN   TestMetric10kDPS/SignalFx
2020/07/08 19:04:20 Starting mock backend...
2020/07/08 19:04:20 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:04:20 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestMetric10kDPS/SignalFx/agent.log
2020/07/08 19:04:20 Agent running, pid=95592
2020/07/08 19:04:20 Starting load generator at 10000 items/sec.
2020/07/08 19:04:23 Agent RAM (RES):  12 MiB, CPU: 0.0% | Sent:   204,400 items | Received:   203,700 items (67,903/sec)
2020/07/08 19:04:26 Agent RAM (RES):  38 MiB, CPU:49.7% | Sent:   414,400 items | Received:   413,700 items (68,950/sec)
2020/07/08 19:04:29 Agent RAM (RES):  38 MiB, CPU:48.0% | Sent:   624,400 items | Received:   623,700 items (69,299/sec)
2020/07/08 19:04:32 Agent RAM (RES):  39 MiB, CPU:49.0% | Sent:   833,700 items | Received:   833,000 items (69,416/sec)
2020/07/08 19:04:35 Agent RAM (RES):  40 MiB, CPU:49.4% | Sent: 1,042,300 items | Received: 1,041,600 items (69,440/sec)
2020/07/08 19:04:35 Stopped generator. Sent: 1,047,200 items
2020/07/08 19:04:35 Gracefully terminating Agent pid=95592, sending SIGTEM...
2020/07/08 19:04:35 Stopping process monitor.
2020/07/08 19:04:35 Agent process stopped, exit code=0
2020/07/08 19:04:35 Sent and received data matches.
2020/07/08 19:04:35 Stopping mock backend...
2020/07/08 19:04:35 Stopped backend. Received: 1,047,200 items (68,480/sec)
--- PASS: TestMetric10kDPS (46.50s)
    --- PASS: TestMetric10kDPS/OpenCensus (15.49s)
    --- PASS: TestMetric10kDPS/Carbon (15.72s)
    --- PASS: TestMetric10kDPS/SignalFx (15.29s)
=== RUN   TestTrace10kSPS
=== RUN   TestTrace10kSPS/OpenCensus
2020/07/08 19:04:35 Starting mock backend...
2020/07/08 19:04:35 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:04:35 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestTrace10kSPS/OpenCensus/agent.log
2020/07/08 19:04:35 Agent running, pid=95701
2020/07/08 19:04:35 Starting load generator at 10000 items/sec.
2020/07/08 19:04:38 Agent RAM (RES):  12 MiB, CPU: 0.0% | Sent:    29,400 items | Received:    27,900 items (9,302/sec)
2020/07/08 19:04:41 Agent RAM (RES):  42 MiB, CPU:25.1% | Sent:    59,400 items | Received:    57,900 items (9,650/sec)
2020/07/08 19:04:44 Agent RAM (RES):  42 MiB, CPU:24.7% | Sent:    89,400 items | Received:    88,200 items (9,800/sec)
2020/07/08 19:04:47 Agent RAM (RES):  42 MiB, CPU:25.0% | Sent:   119,400 items | Received:   118,400 items (9,866/sec)
2020/07/08 19:04:50 Agent RAM (RES):  42 MiB, CPU:24.7% | Sent:   149,400 items | Received:   148,300 items (9,887/sec)
2020/07/08 19:04:50 Stopped generator. Sent:   149,900 items
2020/07/08 19:04:50 Gracefully terminating Agent pid=95701, sending SIGTEM...
2020/07/08 19:04:50 Stopping process monitor.
2020/07/08 19:04:50 Agent process stopped, exit code=0
2020/07/08 19:04:50 Sent and received data matches.
2020/07/08 19:04:50 Stopping mock backend...
2020/07/08 19:04:50 Stopped backend. Received:   149,900 items (9,828/sec)
=== RUN   TestTrace10kSPS/OTLP
2020/07/08 19:04:50 Starting mock backend...
2020/07/08 19:04:50 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:04:50 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestTrace10kSPS/OTLP/agent.log
2020/07/08 19:04:50 Agent running, pid=95794
2020/07/08 19:04:51 Starting load generator at 10000 items/sec.
2020/07/08 19:04:53 Agent RAM (RES):  12 MiB, CPU: 0.0% | Sent:    29,400 items | Received:    28,800 items (9,602/sec)
2020/07/08 19:04:56 Agent RAM (RES):  32 MiB, CPU:15.2% | Sent:    59,400 items | Received:    57,700 items (9,617/sec)
2020/07/08 19:04:59 Agent RAM (RES):  33 MiB, CPU:16.3% | Sent:    89,400 items | Received:    88,600 items (9,845/sec)
2020/07/08 19:05:02 Agent RAM (RES):  33 MiB, CPU:16.3% | Sent:   119,400 items | Received:   117,300 items (9,775/sec)
2020/07/08 19:05:05 Agent RAM (RES):  33 MiB, CPU:16.3% | Sent:   149,400 items | Received:   148,200 items (9,880/sec)
2020/07/08 19:05:06 Stopped generator. Sent:   150,000 items
2020/07/08 19:05:06 Gracefully terminating Agent pid=95794, sending SIGTEM...
2020/07/08 19:05:06 Stopping process monitor.
2020/07/08 19:05:06 Agent process stopped, exit code=0
2020/07/08 19:05:06 Sent and received data matches.
2020/07/08 19:05:06 Stopping mock backend...
2020/07/08 19:05:06 Stopped backend. Received:   150,000 items (9,908/sec)
=== RUN   TestTrace10kSPS/SAPM
2020/07/08 19:05:06 Starting mock backend...
2020/07/08 19:05:06 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:05:06 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestTrace10kSPS/SAPM/agent.log
2020/07/08 19:05:06 Agent running, pid=95908
2020/07/08 19:05:06 Starting load generator at 10000 items/sec.
2020/07/08 19:05:09 Agent RAM (RES):  12 MiB, CPU: 0.0% | Sent:    29,400 items | Received:    27,700 items (9,233/sec)
2020/07/08 19:05:12 Agent RAM (RES):  42 MiB, CPU:20.8% | Sent:    59,400 items | Received:    57,900 items (9,649/sec)
2020/07/08 19:05:15 Agent RAM (RES):  46 MiB, CPU:20.0% | Sent:    89,400 items | Received:    87,900 items (9,767/sec)
2020/07/08 19:05:18 Agent RAM (RES):  47 MiB, CPU:21.0% | Sent:   119,400 items | Received:   118,000 items (9,833/sec)
2020/07/08 19:05:21 Agent RAM (RES):  47 MiB, CPU:21.3% | Sent:   149,400 items | Received:   148,100 items (9,873/sec)
2020/07/08 19:05:21 Stopped generator. Sent:   150,000 items
2020/07/08 19:05:21 Gracefully terminating Agent pid=95908, sending SIGTEM...
2020/07/08 19:05:21 Stopping process monitor.
2020/07/08 19:05:21 Agent process stopped, exit code=0
2020/07/08 19:05:21 Sent and received data matches.
2020/07/08 19:05:21 Stopping mock backend...
2020/07/08 19:05:21 Fatal error reported: http: Server closed
2020/07/08 19:05:21 Stopped backend. Received:   150,000 items (9,842/sec)
=== RUN   TestTrace10kSPS/AwsXray
2020/07/08 19:05:21 Starting mock backend...
2020/07/08 19:05:21 Starting Agent (/Volumes/unix/workplace/opentelemetry-collector-contrib/bin/otelcontribcol_darwin_amd64)
2020/07/08 19:05:21 Writing Agent log to /Volumes/unix/workplace/opentelemetry-collector-contrib/testbed/tests/results/TestTrace10kSPS/AwsXray/agent.log
2020/07/08 19:05:21 Agent running, pid=95994
2020/07/08 19:05:21 Starting load generator at 10000 items/sec.
2020/07/08 19:05:24 Agent RAM (RES):  11 MiB, CPU: 0.0% | Sent:    22,700 items | Received:    22,300 items (7,434/sec)
2020/07/08 19:05:27 Agent RAM (RES):  39 MiB, CPU:39.7% | Sent:    47,600 items | Received:    46,150 items (7,692/sec)
2020/07/08 19:05:30 Agent RAM (RES):  40 MiB, CPU:40.4% | Sent:    72,700 items | Received:    70,300 items (7,811/sec)
2020/07/08 19:05:33 Agent RAM (RES):  41 MiB, CPU:40.0% | Sent:    97,300 items | Received:    95,400 items (7,950/sec)
2020/07/08 19:05:36 Agent RAM (RES):  41 MiB, CPU:39.9% | Sent:   121,800 items | Received:   120,500 items (8,033/sec)
2020/07/08 19:05:36 Stopped generator. Sent:   122,400 items
2020/07/08 19:05:36 Gracefully terminating Agent pid=95994, sending SIGTEM...
2020/07/08 19:05:36 Stopping process monitor.
2020/07/08 19:05:36 Agent process stopped, exit code=1
2020/07/08 19:05:36 Agent execution failed: exit status 1
2020/07/08 19:05:36 Sent and received data matches.
2020/07/08 19:05:36 Stopping mock backend...
2020/07/08 19:05:36 Stopped backend. Received:   122,400 items (7,970/sec)
2020/07/08 19:05:36 Fatal error reported: http: Server closed
--- PASS: TestTrace10kSPS (61.00s)
    --- PASS: TestTrace10kSPS/OpenCensus (15.25s)
    --- PASS: TestTrace10kSPS/OTLP (15.14s)
    --- PASS: TestTrace10kSPS/SAPM (15.24s)
    --- PASS: TestTrace10kSPS/AwsXray (15.36s)
PASS
ok      github.com/open-telemetry/opentelemetry-collector-contrib/testbed/tests 108.816s
# Test PerformanceResults
Started: Wed, 08 Jul 2020 19:03:49 -0700

Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
Metric10kDPS/OpenCensus                 |PASS  |     15s|    24.6|    26.2|         28|         34|   1049300|       1049300|
Metric10kDPS/Carbon                     |PASS  |     16s|   167.0|   170.8|         31|         35|    415100|        415100|
Metric10kDPS/SignalFx                   |PASS  |     15s|    48.9|    49.7|         34|         40|   1047200|       1047200|
Trace10kSPS/OpenCensus                  |PASS  |     15s|    24.7|    25.1|         37|         42|    149900|        149900|
Trace10kSPS/OTLP                        |PASS  |     15s|    16.2|    16.3|         29|         34|    150000|        150000|
Trace10kSPS/SAPM                        |PASS  |     15s|    21.0|    21.3|         40|         48|    150000|        150000|
Trace10kSPS/AwsXray                     |PASS  |     15s|    40.3|    41.1|         35|         41|    122400|        122400|

Total duration: 107s

```

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

File: receiver/awsxrayreceiver/Makefile [hunk 1]
```diff
@@ -0,0 +1 @@
+include ../../Makefile.Common
```
Comment (anuraaga) [inline receiver/awsxrayreceiver/Makefile:1]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/364 starts the work to add a real receiver for xray, how does it relate to that? Can you sync up on it?

Comment (darwhs) [reply receiver/awsxrayreceiver/Makefile:1]: surely I will sync up Yenlinand modify the code accordingly.

**Comment by anuraaga** [comment]
@darwhs Do you think we can close this PR in favor of #502? It looks like currently this PR has a lot of merge conflicts with that code.

**Comment by darwhs** [comment]
> @darwhs Do you think we can close this PR in favor of #502? It looks like currently this PR has a lot of merge conflicts with that code.

yes we should close this PR

**Pull request closed without merging**